azurerm/resource_arm_private_link_service_test.go

```diff
@@ -0,0 +1,303 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMPrivateLinkService_basic(t *testing.T) {
+	resourceName := "azurerm_private_link_service.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateLinkServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateLinkService_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateLinkServiceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "primary_nat_ip_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "load_balancer_frontend_ip_configuration_ids.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "load_balancer_frontend_ip_configuration_ids.0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPrivateLinkService_update(t *testing.T) {
+	resourceName := "azurerm_private_link_service.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateLinkServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateLinkService_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateLinkServiceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "primary_nat_ip_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "load_balancer_frontend_ip_configuration_ids.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "load_balancer_frontend_ip_configuration_ids.0"),
+				),
+			},
+			{
+				Config: testAccAzureRMPrivateLinkService_update(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateLinkServiceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "primary_nat_ip_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "load_balancer_frontend_ip_configuration_ids.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "load_balancer_frontend_ip_configuration_ids.0"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.env", "test"),
+				),
+			},
+			{
```

Comment:
The update test should try and update to a complete version with multiple auxillary ip configs

Reply:
Fixed.